### DISCUSSION TITLE
How to use eventsource?

### DISCUSSION THREAD

**Discussion by zenners** [pocketbase/pocketbase#178] [Q&A] [answered]
Im using Remix and was trying out subscriptions. I got this error as I was using ` client.Realtime.subscribe("demo", function (e) {
    console.log(e.record);
  });`
Got this error
```ReferenceError: EventSource is not defined```

I did ` npm install eventsource ` as per the docs, but not sure how to use it moving forward.

I tried this and it doesn't work 

```
import { client } from "~/db.server";
const EventSource = require('eventsource')

const es = new EventSource(client)

// ??
const es = new EventSource('http://localhost:8090')

```

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure what Remix is, but if it is a node environment, then you'll need a polyfill, eg. https://github.com/EventSource/eventsource.

To load the polyfill, you'll have to define it globally before instantiating the client (similar to the fetch polyfill), aka.:
`global.EventSource = require("eventsource")`.

Here is a simialar React Native issue - https://github.com/pocketbase/js-sdk/issues/14. I'll update the readme later this week with better notes and caveats for node.
